pkg/kubelet/cm/deviceplugin/manager_test.go

```diff
@@ -375,7 +375,7 @@ func makePod(requests v1.ResourceList) *v1.Pod {
 			Containers: []v1.Container{
 				{
 					Resources: v1.ResourceRequirements{
-						Requests: requests,
+						Limits: limits,
```

Comment:
Should we set both requests and limits? That would match the validation change we discussed.

Reply:
Same question for the init containers and regular containers in `TestInitContainerDeviceAllocation` below, which doesn't use this `makePod` function.

Reply:
I think this PR is just a revert patch, maybe we can discuss this in another place :)

Reply:
It isn’t though. I don’t think this diff was created with git revert, because those init containers were touched by the original patch.

Reply:
@ConnorDoyle 
> Should we set both requests and limits? That would match the validation change we discussed.

There is a point in adding requests only if we scan through both, requests and limits, in the code. With this PR, we are scanning limits only. Therefore even if we add requests in the test case pods, those would be simple ignored and will add no value.

Fixed `TestInitContainerDeviceAllocation` to use `limits`. Thanks for pointing that out. Test case was passing with requests because all allocations were coming empty and thus were able to satisfy the `IsSuperset` condition of the test. Fixed this as well by asserting device count in the allocation to make sure allocation is not empty.

Reply:
Btw, for these unit tests it perhaps doesn't matter, but I think in general on kubelet-level unit tests, we probably want to set both Requests and Limits. This matches the behavior that API server will automatically sets Requests equal to Limits when only Limits is set, and on most of the scheduler related code (some of them is also run by kubelet during admission), we look at Requests.